lib/ansible/module_utils/network/vyos/facts/lag_interfaces/lag_interfaces.py

```diff
@@ -0,0 +1,132 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The vyos lag_interfaces fact class
+It is in this file the configuration is collected from the device
+for a given resource, parsed, and the facts tree is populated
+based on the configuration.
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+from re import findall, search, M
+from copy import deepcopy
+
+from ansible.module_utils.network.common import utils
+from ansible.module_utils.network.vyos.argspec.lag_interfaces. \
+    lag_interfaces import Lag_interfacesArgs
+
+
+class Lag_interfacesFacts(object):
+    """ The vyos lag_interfaces fact class
+    """
+
+    def __init__(self, module, subspec='config', options='options'):
+        self._module = module
+        self.argument_spec = Lag_interfacesArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for lag_interfaces
+        :param module: the module instance
+        :param connection: the device connection
+        :param data: previously collected conf
+        :rtype: dictionary
+        :returns: facts
+        """
+        if connection:  # just for linting purposes, remove
+            pass
+
+        if not data:
+            data = connection.get_config()
+
+        objs = []
+        lag_names = findall(r'^set interfaces bonding (\S+)', data, M)
+        if lag_names:
+            for lag in set(lag_names):
+                lag_regex = r' %s .+$' % lag
+                cfg = findall(lag_regex, data, M)
+                obj = self.render_config(cfg)
+
+                output = connection.run_commands(['show interfaces bonding ' + lag + ' slaves'])
+                lines = output[0].splitlines()
+                members = []
+                member = {}
+                if len(lines) > 1:
+                    for line in lines[2:]:
+                        splitted_line = line.split()
+
+                        if len(splitted_line) > 1:
+                            member['member'] = splitted_line[0]
+                            members.append(member)
+                        else:
+                            members = []
+                        member = {}
+                obj['name'] = lag.strip("'")
+                if members:
+                    obj['members'] = members
+
+                if obj:
+                    objs.append(obj)
+
+        facts = {}
+        if objs:
+            facts['lag_interfaces'] = objs
```

Comment:
`validate_config` call is missing, IMO that should be added. Please refer to [this](https://github.com/ansible/network/pull/42/files#diff-40a2b0039c14ce8fc94b0244e9ce8614R61).